app/policies/custom_filter_policy.rb

```diff
@@ -0,0 +1,21 @@
+class CustomFilterPolicy < ApplicationPolicy
+  def create?
+    @account_user.administrator? || @account_user.agent?
+  end
+
+  def show?
+    @account_user.administrator? || @account_user.agent?
+  end
+
+  def index?
+    @account_user.administrator? || @account_user.agent?
+  end
+
+  def update?
+    @account_user.administrator? || @account_user.agent?
+  end
+
```

Comment:
we could move the logic  that `agent x` is not able to view filters created by `agent y`  etc here in future. the policy file seems like the right place for that